src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/InfoTests.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Monitoring.UnitTests.Fixtures;
+using Microsoft.Diagnostics.Monitoring.UnitTests.HttpApi;
+using Microsoft.Diagnostics.Monitoring.UnitTests.Models;
+using Microsoft.Diagnostics.Monitoring.UnitTests.Options;
+using Microsoft.Diagnostics.Monitoring.UnitTests.Runners;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Diagnostics.Monitoring.UnitTests
+{
+    [Collection(DefaultCollectionFixture.Name)]
+    public class InfoTests
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly ITestOutputHelper _outputHelper;
+
+        public InfoTests(ITestOutputHelper outputHelper, ServiceProviderFixture serviceProviderFixture)
+        {
+            _httpClientFactory = serviceProviderFixture.ServiceProvider.GetService<IHttpClientFactory>();
+            _outputHelper = outputHelper;
+        }
+
+        /// <summary>
+        /// Tests that the info endpoint provides the expected output.
+        /// </summary>
+        [Theory]
+        [InlineData(DiagnosticPortConnectionMode.Connect)]
+#if NET5_0_OR_GREATER
+        [InlineData(DiagnosticPortConnectionMode.Listen)]
+#endif
+        public Task InfoEndpointValidationTest(DiagnosticPortConnectionMode mode)
+        {
+            return ScenarioRunner.SingleTarget(
+                _outputHelper,
+                _httpClientFactory,
+                mode,
+                TestAppScenarios.AsyncWait.Name,
+                appValidate: async (runner, client) =>
+                {
+                    // GET /info
+                    DotnetMonitorInfo info = await client.GetInfoAsync();
+
+                    Assert.NotNull(info.Version); // Not sure of how to get Dotnet Monitor version from within tests...
```

Comment:
While this is not a great option, but do we uniformly version all the built assemblies? If so, then the test assembly version and the dotnet-monitor version are likely to be the same. Another option would be to load dotnet-monitor.dll into the test process but I don't think we should go that far.

Reply:
Are you saying that if we get the version the way we do in DiagController (GetDotnetMonitorVersion()) that it might be the same? That's what I originally experimented with, and the test failed saying that version was '5.0.0-dev...' vs '16.6.1'. Was this what you had in mind, or were you suggesting a different approach?